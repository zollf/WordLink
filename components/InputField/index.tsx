import React, { useState } from "react";
import { TextInput } from "react-native";

import { widthPercentageToDP } from "react-native-responsive-screen";

import InputFieldStyles from "./styles";
import AppStyle from "../../styles";

interface Props {
  color: "orange" | "blue";
  staticValue: string;
  size?: "small" | "full";
}

const InputField = ({ color, staticValue = "", size = "full" }: Props) => {
  const [value, setValue] = useState("");

  const sizes = {
    full: "60%",
    small: "27.5%",
  };
  return (
    <TextInput
      data-test-id="text-input"
      editable={!staticValue}
      selectTextOnFocus={!staticValue}
      onChange={(e) => setValue(e.nativeEvent?.text)}
      value={staticValue || value}
      style={[
        InputFieldStyles.input,
        {
          width: widthPercentageToDP(sizes[size]),
          backgroundColor: AppStyle[color],
        },
      ]}
    />
  );
};

export default InputField;
