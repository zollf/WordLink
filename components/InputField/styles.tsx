import { StyleSheet } from "react-native";

import { heightPercentageToDP } from "react-native-responsive-screen";
import AppStyle from "../../styles";

const InputFieldStyles = StyleSheet.create({
  gradient: {
    height: heightPercentageToDP("7%"),
    borderRadius: 5,
    shadowColor: "#000",
    shadowRadius: 4,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    color: "white",
    fontSize: heightPercentageToDP("4%"),
    fontFamily: AppStyle.fontBold,
    textAlign: "center",
  },
});
export default InputFieldStyles;
