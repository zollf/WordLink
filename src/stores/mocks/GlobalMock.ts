const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
