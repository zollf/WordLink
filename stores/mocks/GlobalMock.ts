const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
