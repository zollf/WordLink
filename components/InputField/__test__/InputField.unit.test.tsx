import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';

import InputField from '../';

import renderer, { act } from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock}>
    <InputField color="orange" value="" />
  </Provider>
);

describe('<InputField />', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('handles change correctly', async () => {
    await act(async () => {
      wrapper.root.findByProps({ 'data-test-id': 'text-input' }).props.onChange('event', {
        nativeElement: {
          text: 'test',
        },
      });
    });
    //todo testing
  });
});
