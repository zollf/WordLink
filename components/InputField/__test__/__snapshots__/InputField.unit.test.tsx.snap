// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> matches snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  data-test-id="text-input"
  onChange={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "height": 39,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
        "textAlign": "center",
        "width": 229,
      },
      Object {
        "backgroundColor": "#FF9C40",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="SampleUsername"
/>
`;
